holidays/countries/thailand.py

```diff
@@ -317,12 +339,28 @@ def _add_observed(dt: date) -> None:
 
         new_years_eve_in_lieu = self.tr("ชดเชย%s") % self.tr("วันสิ้นปี")
 
-        if self.observed and (1995 <= year <= 1997 or year >= 2001):
+        if self.observed and (1995 <= self._year <= 1997 or self._year >= 2001):
             if self._is_sunday(jan_1):
                 self._add_new_years_day_three(new_years_eve_in_lieu)
             elif self._is_monday(jan_1):
                 self._add_new_years_day_two(new_years_eve_in_lieu)
 
+        # วันเด็กแห่งชาติ
+        # Status: In-Use.
+        # Starts in 1955 as the 1st Monday of October.
+        # No event was held in 1964 due to date changes came into effect too late.
+        # Moved to 2nd Saturday of January since 1965.
+        # No in-lieus are observed, and still remain a Public Holidays than just Observed.
+
+        if self._year >= 1955 and self._year != 1964:
+            self._add_holiday(
+                # National Children's Day
+                tr("วันเด็กแห่งชาติ"),
+                self._get_nth_weekday_of_month(1, MON, OCT)
```

Comment:
Now we need to use `_get_nth_weekday_of_month()` from `holidays.calendars.gregorian` instead of method.

Reply:
Fixed.